dotCMS/src/main/java/com/dotcms/rest/api/v1/system/monitor/MonitorResource.java

```diff
@@ -0,0 +1,343 @@
+                CACHE_TIMEOUT = Config.getLongProperty("SYSTEM_STATUS_API_CACHE_TIMEOUT", 1000);
+                ASSET_FS_TIMEOUT = Config.getLongProperty("SYSTEM_STATUS_API_ASSET_FS_TIMEOUT", 1000);
+                INDEX_TIMEOUT = Config.getLongProperty("SYSTEM_STATUS_API_INDEX_TIMEOUT", 1000);
+                DB_TIMEOUT = Config.getLongProperty("SYSTEM_STATUS_API_DB_TIMEOUT", 1000);
+
+                boolean dotCMSHealthy = false;
+                boolean frontendHealthy = false;
+                boolean backendHealthy = false;
+                final boolean dbSelectHealthy = dbCount();
+                final boolean indexLiveHealthy = indexCount(idxs.live);
+                final boolean indexWorkingHealthy = indexCount(idxs.working);
+                final boolean cacheHealthy = cache();
+                final boolean localFSHealthy = localFiles();
+                final boolean assetFSHealthy = assetFiles();
+
+                if (dbSelectHealthy && indexLiveHealthy && indexWorkingHealthy && cacheHealthy && localFSHealthy && assetFSHealthy) {
+                    dotCMSHealthy = true;
+                    frontendHealthy = true;
+                    backendHealthy = true;
+                } else if (!indexWorkingHealthy && dbSelectHealthy && indexLiveHealthy && cacheHealthy && localFSHealthy && assetFSHealthy) {
+                    frontendHealthy = true;
+                }
+
+                if (extendedFormat) {
+                    final String serverID = getServerID();
+                    final String clusterID = getClusterID();
+                    final JSONObject jo = new JSONObject();
+                    jo.put("serverID", serverID);
+                    jo.put("clusterID", clusterID);
+                    jo.put("dotCMSHealthy", dotCMSHealthy);
+                    jo.put("frontendHealthy", frontendHealthy);
+                    jo.put("backendHealthy", backendHealthy);
+
+                    final JSONObject subsystems = new JSONObject();
+                    subsystems.put("dbSelectHealthy", dbSelectHealthy);
+                    subsystems.put("indexLiveHealthy", indexLiveHealthy);
+                    subsystems.put("indexWorkingHealthy", indexWorkingHealthy);
+                    subsystems.put("cacheHealthy", cacheHealthy);
+                    subsystems.put("localFSHealthy", localFSHealthy);
+                    subsystems.put("assetFSHealthy", assetFSHealthy);
+                    jo.put("subsystems", subsystems);
+
+                    builder = Response.ok(jo.toString(2), MediaType.APPLICATION_JSON);
+                } else {
+                    if (dotCMSHealthy) {
+                        builder = Response.ok("", MediaType.APPLICATION_JSON);
+                    } else if (!indexWorkingHealthy && dbSelectHealthy && indexLiveHealthy && cacheHealthy && localFSHealthy && assetFSHealthy) {
+                        builder = Response.status(507).entity("").type(MediaType.APPLICATION_JSON);
```

Comment:
we already have constants for the response code, also a message should be returned, we already have helpers and utils for it